ExternalModule.php

```diff
@@ -28,74 +28,85 @@ function hook_survey_complete($project_id, $record, $instrument, $event_id, $gro
       $source_instruments = AbstractExternalModule::getProjectSetting('ssptf_source_instrument');
 
       //check if instrument is the same one set in config
-      $index = array_search($instrument, $source_instruments);
+      $indices = array_keys($source_instruments, $instrument);
 
       //abort hook if not
-      if($index === FALSE) {
+      if($indices === FALSE) {
         return 0;
       }
 
-      //get target upload field from config
-      $target_fields = AbstractExternalModule::getProjectSetting('ssptf_target_upload_field');
-      $target_upload_field = $target_fields[$index];
+      // Count number of matched targets
+      $target_cnt=count($indices);
 
-      $matches = array();
-      $index = 1;
-      $target_upload_field_name = $target_upload_field;
-      if (preg_match('#^(.+)_([0-9]+)$#', $target_upload_field, $matches)) {
-          $target_upload_field_name = $matches[1];
-          $index = (int)$matches[2];
-      }
+      // Iterate along dictionary of targeted keys
+      for($target = 0; $target <= $target_cnt-1; $target++) {
```

Comment:
maybe instead use 
`foreach ( $indices as $target=>$index) {`

Reply:
Addressed in 9d602b4